pkg/tfshim/sdk-v2/provider_options.go

```diff
@@ -49,3 +50,11 @@ func getProviderOptions(opts []providerOption) (providerOptions, error) {
 	}
 	return res, nil
 }
+
+// Selectively opt-in resources that pass the filter to using PlanResourceChange.
+func WithPlanResourceChange(filter func(string) bool) providerOption {
```

Comment:
Does filter take a tf token or a pulumi token?

```suggestion
func WithPlanResourceChange(filter func(string) bool) providerOption {
```